core/src/test/scala/unit/kafka/tools/DumpLogSegmentsTest.scala

```diff
@@ -598,6 +598,25 @@ class DumpLogSegmentsTest {
     )
   }
 
+  @Test
+  def testNewDecoder(): Unit = {
+    // Decoder translate should pass without exception
+    DumpLogSegments.newDecoder(classOf[DumpLogSegmentsTest.TestDecoder].getName)
```

Comment:
Could you add UT to make sure the impl of deprecated decoded must have constructor with `VerifiableProperties`